src/transforms.rs

```diff
@@ -120,23 +120,23 @@ pub fn reduce_authors<'a, I>(rec: Vec<CSLValue>) -> Vec<CSLValue> {
             .map!(y => y.takeOne)
         ).joiner.joiner;
     */
-    let mut reduced = names_grouped_by_type
+    let reduced = names_grouped_by_type
         .map(|n| n.group_by(
-            |a, b|  a.np().unwrap().family.unwrap().split(" ").nth(0) ==
-                    b.np().unwrap().family.unwrap().split(" ").nth(0))
-
-        );
+            |a, b|  a.np().unwrap().family.as_ref().unwrap().split(" ").nth(0) ==
+                    b.np().unwrap().family.as_ref().unwrap().split(" ").nth(0))
+            .map(|y| y[0])
+        ).flatten();
     // `reduced` now contains deduplicated names
 
-    let mut no_names = rec.iter().filter(|x| !x.is_name());
+    let no_names = rec.iter().filter(|x| !x.is_name());
     // D:   return chain(noNames, reduced);
-    no_names.chain(reduced).collect()
+    no_names.chain(reduced).map(|x| x.clone()).collect()
```

Comment:
Why is the `map(|x| x.clone())` required, and is there a more ergonomic way to do this?

Reply:
The clone is required as the result of `chain` is actually an iterator of `&CSLValue` (so `collect` would make it a `Vec<&CSLValue>`) which conflicts with the return type of `Vec<CSLValue>`. I could see if we can return `Vec<&CSLValue>` if you would like to avoid the clone.

Reply:
The reason `chain` is returning an iterator of `&CSLValue` is because that is what `group_by` returns.